crates/orchestrator/src/jobs/state_update_job/mod.rs

```diff
@@ -165,7 +165,11 @@ impl Job for StateUpdateJob {
         // number : 0
         let metadata_tx_hashes = job
             .metadata
-            .get(&format!("{}{}", JOB_METADATA_STATE_UPDATE_ATTEMPT_PREFIX, attempt_no.parse::<u32>().unwrap() - 1))
+            .get(&format!(
+                "{}{}",
+                JOB_METADATA_STATE_UPDATE_ATTEMPT_PREFIX,
+                attempt_no.parse::<u32>().map_err(|e| JobError::Other(OtherError(eyre!(e))))? - 1
```

Comment:
tbh this other error thing is getting a little ugly at this point. will need to think of a more elegant solution but that's not the scope of this PR. can you create an issue so we can track this?